### ISSUE TITLE
maven publish: env variables enforced that are not needed

### ISSUE THREAD

**Issue by webratz** [projen/projen#1161]
Hey,

I'm not entirely sure if this is the right place to report the issue. Let me know if it should be elsewhere.

The generated `publish:maven` step (i assume its coming from: https://github.com/projen/projen/blob/63f89f0d7adbc853a2caeac16de63cac75e257be/src/release/publisher.ts#L246)
is generating a section in `tasks.json` like this:

```json
    "publish:maven": {
      "name": "publish:maven",
      "description": "Publish this package to Maven Central",
      "env": {
        "MAVEN_ENDPOINT": "https://myscustomurl.com",
        "MAVEN_REPOSITORY_URL": "https://myscustomurl.com"
      },
      "requiredEnv": [
        "MAVEN_GPG_PRIVATE_KEY",
        "MAVEN_GPG_PRIVATE_KEY_PASSPHRASE",
        "MAVEN_PASSWORD",
        "MAVEN_USERNAME",
        "MAVEN_STAGING_PROFILE_ID"
      ],
      "steps": [
        {
          "exec": "npx -p jsii-release@latest jsii-release-maven"
        }
      ]
    },
```
This results in
```
🤖 missing required environment variables: MAVEN_GPG_PRIVATE_KEY,MAVEN_GPG_PRIVATE_KEY_PASSPHRASE,MAVEN_PASSWORD,MAVEN_USERNAME,MAVEN_STAGING_PROFILE_ID
```
when running `npx projen publish:maven`
This does not work for repositories where not maven central is used, and only there the GPG signing is supported.

the publish works perfectly fine when setting the env vars manually, eg like that:
```bash
export MAVEN_ENDPOINT="https://myscustomurl.comrepository/maven/",
export MAVEN_REPOSITORY_URL="https://myscustomurl.com/repository/maven-releases/"
export MAVEN_SERVER_ID="nexus"
export MAVEN_USERNAME="foobar"
export MAVEN_PASSWORD="CHANGEME"

npx -p jsii-release@latest jsii-release-maven
```

But fails as soon any of the GPG variables is set:
```bash
export MAVEN_GPG_PRIVATE_KEY="sfhgjk"

❌ Package signing is currently only supported for Maven Central
```
projen version: 0.30.5

tl;dr:

Is there an option to not make the gpg env variables mandatory for the npm publish step?
